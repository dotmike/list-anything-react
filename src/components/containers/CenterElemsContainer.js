import React from "react";
import PropTypes from "prop-types";
import { Grid } from "semantic-ui-react";

const CenterElemsContainer = ({ children }) => (
  <div className="form-container">
    <style>{`
      body > div,
      body > div > div,
      body > div > div > div.form-container {
          height: 100%;
      }
    `}</style>
    <Grid textAlign="center" style={{ height: "100%" }} verticalAlign="middle">
      <Grid.Column style={{ maxWidth: 450 }}>{children}</Grid.Column>
    </Grid>
  </div>
);

CenterElemsContainer.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired
};

export default CenterElemsContainer;
