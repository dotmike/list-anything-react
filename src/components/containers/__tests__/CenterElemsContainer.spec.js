import React from "react";
import { shallow } from "enzyme";

import CenterElemsContainer from "../CenterElemsContainer";

describe("CenterElemsContainer", () => {
  it("renders correctly", () => {
    const wrapper = shallow(
      <CenterElemsContainer>Testando...</CenterElemsContainer>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
