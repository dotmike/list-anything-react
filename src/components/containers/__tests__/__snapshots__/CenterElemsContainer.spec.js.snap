// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CenterElemsContainer renders correctly 1`] = `
<Grid
  style={
    Object {
      "height": "100%",
    }
  }
  textAlign="center"
  verticalAlign="middle"
>
  <GridColumn
    style={
      Object {
        "maxWidth": 450,
      }
    }
  >
    Testando...
  </GridColumn>
</Grid>
`;
